### PR TITLE
34 shutdown hook

### PR DESCRIPTION
**Pull request by Viktormes**
Add shutdown hook and todo:s.

### CODE REVIEW

**Comment by BobHogvall** [comment]
From what I can see, you might want to try and create your own method for throwing an exception. So instead of using the 'generic' ones given to you, create your own one? 

Regards <3

**Comment by BobHogvall** [comment]
https://www.baeldung.com/java-new-custom-exception

**Review by Svartkettil — COMMENTED** [review]
It seems like we need to throw an exception that is non generic. Maybe this could be an issue of its own? To make our own runTimeException?

File: src/main/java/org/fungover/haze/Main.java [hunk 1]
```diff
@@ -46,13 +57,21 @@ public static void main(String[] args) throws IOException {
 
                     } catch (IOException e) {
                         Log4j2.error(String.valueOf(e));
+                        throw new RuntimeException(e);
```
Comment (jLereback) [inline src/main/java/org/fungover/haze/Main.java:60]:
```suggestion
```

**Review by jLereback — CHANGES_REQUESTED** [review]
There is an extra exception that we don't need :hugs:

**Review by xCandyFun — APPROVED** [review]
Looking good

**Review by safstromo — APPROVED** [review]
:100:

**Review by jLereback — APPROVED** [review]
Great last commit :smile:

**Review by cchriss123 — APPROVED** [review]
Code looks great now

**Review by RobertMili — APPROVED** [review]

**Comment by jLereback** [comment]
Added in #133

**Pull request closed without merging**